File: backend/src/main/java/dev/voroby/client/chatList/dto/ChatPreview.java
package dev.voroby.client.chatList.dto;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record ChatPreview(
        long id,
        String title,
        String photo,
        String lastMessage,
        long unreadCount,
        long order,
        ChatType chatType,
        boolean isChannel,
        boolean canSendTextMessage
) {}

File: desktop/src/main/kotlin/scene/dto/ChatPreview.kt
package scene.dto

data class ChatPreview(
    val id: Long = -1,
    val title: String,
    val photo: String?,
    val lastMessage: String,
    val unreadCount: Int?,
    val order: Long,
    val chatType: ChatType,
    val isChannel: Boolean = false,
    val canSendTextMessage: Boolean = false
)
